research/05-problem-ana.tex

```diff
@@ -1,10 +1,127 @@
 \section{Problem Analysis}
 \label{sec:problem-analysis}
+This section gives an analysis of the problem as defined in the
+previous section \ref{sec:problem-definition}. More specifically, this
+section describes the problems that are expected and to be solved
+during the implementation process of the product. The first part of
+this section describes the process of incorporating REPL generation
+within the larger context of Spoofax. The second part analyzes the
+part of the problem concerned with literate programming, but this part
+will be analyzed in less detail than generating REPLs as it is unsure
+whether this will be present in the final product.
 
-\subsection{How does REPL generation fit within Spoofax}
-\label{ssec:architecture}
+\subsection{Incorporating REPL generation within Spoofax}
+\label{ssec:incorp-repl-gener}
+To generate solidly working REPLs for any language defined in Spoofax,
+some Spoofax concepts have to be carefully considered. Since every
+language can have many different language constructs, generating REPLs
+should not be constrained to language specific constructs like
+classes, but should rather be done generically. Therefore the
+implementation and particularly the interaction with the user needs to
+be thought of carefully.
 
-\subsection{How are plugins developed for Eclipse?}
+This section lists the problems that are expected when trying to
+implement generating REPLs generically. For each problem, their
+solution approaches that should be explored during the implementation
+process are given, as well as the parts of Spoofax that are relevant
+for that problem.
+
+\subsubsection{Detecting unfinished expressions for multiline editing}
+\label{sec:detect-unfin-expr}
+To support multiline editing, the generated REPL should detect that
+the expression is unfinished when the user presses ``enter'' for the
+next line, instead of trying to parse and execute the unfinished
+expression. An obvious part of Spoofax that is relevant for this
+problem, is the syntax definition in SDF3 (\cref{sec:orgheadline1}).
+
+\subsubsection{Language specific additional commands}
```

Comment:
This is talking about exactly the same thing, commands for the REPL that make sense only for a specific language.